internal/common/database/db_testutil.go

```diff
@@ -8,48 +8,65 @@ import (
 	"github.com/jackc/pgx/v4/pgxpool"
 	"github.com/pkg/errors"
 
+	"github.com/G-Research/armada/internal/armada/configuration"
 	"github.com/G-Research/armada/internal/common/util"
 )
 
-func WithTestDb(migrations []migration, action func(db *pgxpool.Pool) error) error {
+// WithTestDb spins up a Postgres database for testing
+// migrations: perform the list of migrations before entering the action callback
+// configOverride: optional PostgresConfig to specify which instance to connect to. Defaults to localhost
+// note - if an override is specified, the database will not be cleaned up after the test
+// action: callback for client code
+func WithTestDb(migrations []Migration, configOverride *configuration.PostgresConfig, action func(db *pgxpool.Pool) error) error {
```

Comment:
I'd probably have two functions, one that takes an override, one that doesn't (on the grounds that most unit tests won't want an override and passing a random nil everywhere is a bit ugly)